src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1253,3 +1253,104 @@ def get_parent_proc_name():
         parent_proc_name = _get_parent_proc_name()
         setattr(get_parent_proc_name, "return_value", parent_proc_name)
     return getattr(get_parent_proc_name, "return_value")
+
+
+def log_cmd_history(command, args):
+    import os
+    from knack.util import ensure_dir
+    from azure.cli.core.extension import get_extension, ExtensionNotInstalledException
+    from azure.cli.core._environment import get_config_dir
+
+    if command == 'next':
+        return
+
+    # Determine whether "az next" has been installed.
+    # At present, command execution log is only recorded when "az next" is installed
+    try:
+        az_next_is_installed = get_extension("next")
+    except ExtensionNotInstalledException:
+        az_next_is_installed = False
+
+    if not az_next_is_installed:
+        return
+
+    # For commands that do not require logging, we can pass in the parameter "--no-log" to control
+    if '--no-log' in args:
+        return
+
+    if not args or not command:
+        return
+
+    base_dir = os.path.join(get_config_dir(), 'recommendation')
+    ensure_dir(base_dir)
+
+    file_path = os.path.join(base_dir, 'cmd_history.log')
+    if not os.path.exists(file_path):
+        with open(file_path, 'w') as fd:
+            fd.write('')
+
+    lines = []
+    with open(file_path, 'r') as fd:
+        lines = fd.readlines()
+        lines = [x.strip('\n') for x in lines if x]
+
+    with open(file_path, 'w') as fd:
+        command_info = {'command': command}
+        params = []
+        for arg in args:
+            if arg.startswith('-'):
+                params.append(arg)
+        if params:
+            command_info['arguments'] = params
+
+        lines.append(json.dumps(command_info))
+        if len(lines) > 15:
+            lines = lines[-15:]
```

Comment:
why we only choose last 15 lines here?

Reply:
Because I had concern that too many commands will be recorded, which may occupy the user's disk space. At present, we actually only use the last two commands for analysis

Reply:
If we could record command scenario, it would be more useful for analysis. 
Current design is fine for start stage, but hope we could consider the improvement later.

Reply:
Yes~ At present, there are technical challenges in locally determining whether a group of commands are in a scenario, we will consider this optimization in our long-term plan